packages/link-modules/src/services/link-module-service.ts

```diff
@@ -100,16 +100,19 @@ export default class LinkModuleService<TLink> implements ILinkModule {
     return this.primaryKey_.concat(this.foreignKey_).includes(name)
   }
 
-  private validateFields(data: any) {
-    const keys = Object.keys(data)
-    if (!keys.every((k) => this.isValidKeyName(k))) {
-      throw new MedusaError(
-        MedusaError.Types.INVALID_DATA,
-        `Invalid field name provided. Valid field names are ${this.primaryKey_.concat(
-          this.foreignKey_
-        )}`
-      )
-    }
+  private validateFields(data: any | any[]) {
+    const dataToValidate = Array.isArray(data) ? data : [data]
+    dataToValidate.forEach((d) => {
+      const keys = Object.keys(d)
+      if (!keys.every((k) => this.isValidKeyName(k))) {
```

Comment:
NIT: `keys.some(k => !this.isValidName(k))`, it's just a tiny bit easier to read